app/com/arpnetworking/kairos/client/models/MetricsQueryResponse.java

```diff
@@ -0,0 +1,309 @@
+
+        private QueryResult(final Builder builder) {
+            _other = builder._other;
+            _values = builder._values;
+        }
+
+        private final Map<String, Object> _other;
+        private final List<DataPoint> _values;
+
+        /**
+         * Implementation of the builder pattern for a {@link QueryResult}.
+         *
+         * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+         */
+        public static final class Builder extends OvalBuilder<QueryResult> {
+            /**
+             * Public constructor.
+             */
+            public Builder() {
+                super(QueryResult::new);
+            }
+
+            /**
+             * Adds an "unknown" parameter. Optional.
+             *
+             * @param key key for the entry
+             * @param value value for the entry
+             * @return this {@link Builder}
+             */
+            @JsonAnySetter
+            public Builder setOther(final String key, final Object value) {
+                _other.put(key, value);
+                return this;
+            }
+
+            /**
+             * Sets the values list. Optional. Cannot be null.
+             *
+             * @param value the values
+             * @return this {@link Builder}
+             */
+            public Builder setValues(final List<DataPoint> value) {
+                _values = value;
+                return this;
+            }
+
+            @NotNull
+            private List<DataPoint> _values = Lists.newArrayList();
+            @NotNull
+            private Map<String, Object> _other = Maps.newHashMap();
+        }
+    }
+
+    /**
+     * Model class for a data point in a kairosdb metrics query.
+     *
+     * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+     */
+    public static final class DataPoint {
+        public DateTime getTime() {
+            return _time;
+        }
+
+        public Object getValue() {
+            return _value;
+        }
+
+        @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Jackson
+        @JsonValue
+        private List<Object> serialize() {
+            return Lists.newArrayList(_time.getMillis(), _value);
+        }
+
+        private DataPoint(final Builder builder) {
+            _time = builder._time;
+            _value = builder._value;
+        }
+
+        private final DateTime _time;
+        private final Object _value;
+
+        /**
+         * Implementation of the builder pattern for a {@link DataPoint}.
+         *
+         * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+         */
+        public static final class Builder extends OvalBuilder<DataPoint> {
+            /**
+             * Public constructor.
+             *
+             * @param arr a 2-element {@link List} containing the time and value at that time.
+             */
+            @JsonCreator
+            public Builder(final List<Object> arr) {
```

Comment:
We should still provide a completed builder so it can be used in testing.

Reply:
Sure.